File: src/Walrhouse.Application/Batches/Queries/GetBatches/BatchDto.cs
using Walrhouse.Domain.Entities;
using Walrhouse.Domain.Enums;

namespace Walrhouse.Application.Batches.Queries.GetBatches;

public class BatchDto
{
    public string BatchNumber { get; set; } = string.Empty;
    public string ItemCode { get; set; } = string.Empty;
    public DateTimeOffset ExpiryDate { get; set; }
    public BatchStatus Status { get; set; }
    public string BinNo { get; set; } = string.Empty;

    public class Mapping : Profile
    {
        public Mapping()
        {
            CreateMap<Batch, BatchDto>();
        }
    }
}

File: src/Walrhouse.Web/App/WebClient/src/features/receive/types/batch-dto.ts
export interface BatchDto {
  batchNumber: string;
  itemCode: string;
  expiryDate: string; // ISO date string
  status: number; // Enum: 0=Pending, 1=Active, 2=Expired, 3=Closed, 4=Inactive
  binNo: string;
}
